src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/HtmlHelper.cs

```diff
@@ -755,7 +753,7 @@ protected virtual string GenerateDisplayText(ModelExplorer modelExplorer)
                 htmlAttributes);
             if (tagBuilder != null)
             {
-                ViewContext.Writer.Write(tagBuilder.ToString(TagRenderMode.StartTag));
+                tagBuilder.ToHtmlContent(TagRenderMode.StartTag).WriteTo(ViewContext.Writer, HtmlEncoder);
```

Comment:
Bet this and line 798 are the only remaining uses of the `HtmlEncoder` property. If so, remove it from the `IHtmlHelper` interface and make it a `private _htmlEncoder` field instead. (Nothing similar to this property exists in MVC 5 and we shouldn't make it easy to encode outside `IHtmlContent.WriteTo()`.)